In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import pickle
import numpy as np
from pandas_profiling import ProfileReport
%matplotlib inline

In [2]:
df = pd.read_csv("https://raw.githubusercontent.com/justmarkham/scikit-learn-videos/master/data/Advertising.csv")
df.head()

Unnamed: 0     TV  Radio  Newspaper  Sales
0           1  230.1   37.8       69.2   22.1
1           2   44.5   39.3       45.1   10.4
2           3   17.2   45.9       69.3    9.3
3           4  151.5   41.3       58.5   18.5
4           5  180.8   10.8       58.4   12.9

In [3]:
df.shape

(200, 5)

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  200 non-null    int64  
 1   TV          200 non-null    float64
 2   Radio       200 non-null    float64
 3   Newspaper   200 non-null    float64
 4   Sales       200 non-null    float64
dtypes: float64(4), int64(1)
memory usage: 7.9 KB


In [5]:
df.describe()

Unnamed: 0          TV       Radio   Newspaper       Sales
count  200.000000  200.000000  200.000000  200.000000  200.000000
mean   100.500000  147.042500   23.264000   30.554000   14.022500
std     57.879185   85.854236   14.846809   21.778621    5.217457
min      1.000000    0.700000    0.000000    0.300000    1.600000
25%     50.750000   74.375000    9.975000   12.750000   10.375000
50%    100.500000  149.750000   22.900000   25.750000   12.900000
75%    150.250000  218.825000   36.525000   45.100000   17.400000
max    200.000000  296.400000   49.600000  114.000000   27.000000

In [6]:
pf = ProfileReport(df)
pf.to_widgets()

Summarize dataset:   0%|          | 0/18 [00:00<?, ?it/s]

Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render widgets:   0%|          | 0/1 [00:00<?, ?it/s]

In [7]:
pf.to_file("test.html")

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

In [8]:
x = df[["TV"]]
y = df[["Sales"]]

In [9]:
from sklearn.linear_model import LinearRegression
lr = LinearRegression()

In [10]:
lr.fit(x, y)

LinearRegression()

In [11]:
lr.intercept_ # c value

array([7.03259355])

In [12]:
lr.coef_ # m value

array([[0.04753664]])

In [13]:
file = "linear_reg.sav"
pickle.dump(lr, open(file, 'wb'))

In [14]:
lr.predict([[45]])

/home/calkikhunt/.local/lib/python3.8/site-packages/sklearn/base.py:450: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


array([[9.17174237]])

In [15]:
l = [4,5,89,40]
for i in l:
    print(lr.predict([[i]]))

[[7.22274011]]
[[7.27027675]]
[[11.26335455]]
[[8.93405917]]


/home/calkikhunt/.local/lib/python3.8/site-packages/sklearn/base.py:450: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(
/home/calkikhunt/.local/lib/python3.8/site-packages/sklearn/base.py:450: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(
/home/calkikhunt/.local/lib/python3.8/site-packages/sklearn/base.py:450: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(
/home/calkikhunt/.local/lib/python3.8/site-packages/sklearn/base.py:450: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


In [16]:
saved_model = pickle.load(open("linear_reg.sav","rb"))

In [17]:
saved_model

LinearRegression()

In [18]:
saved_model.predict([[45]])

/home/calkikhunt/.local/lib/python3.8/site-packages/sklearn/base.py:450: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


array([[9.17174237]])

In [19]:
saved_model.score(x, y) # acurracy
# do not try to pass data which is being used during training 

0.611875050850071

In [20]:
df.head()

Unnamed: 0     TV  Radio  Newspaper  Sales
0           1  230.1   37.8       69.2   22.1
1           2   44.5   39.3       45.1   10.4
2           3   17.2   45.9       69.3    9.3
3           4  151.5   41.3       58.5   18.5
4           5  180.8   10.8       58.4   12.9

In [21]:
data_x = df[["TV", "Radio", "Newspaper"]]
data_x.head()

TV  Radio  Newspaper
0  230.1   37.8       69.2
1   44.5   39.3       45.1
2   17.2   45.9       69.3
3  151.5   41.3       58.5
4  180.8   10.8       58.4

In [22]:
y = df[["Sales"]]
y.head()

Sales
0   22.1
1   10.4
2    9.3
3   18.5
4   12.9

In [23]:
lr2 = LinearRegression()
lr2.fit(data_x, y)

LinearRegression()

In [24]:
lr2.intercept_

array([2.93888937])

In [25]:
lr2.coef_

array([[ 0.04576465,  0.18853002, -0.00103749]])

y = mx + c<br><br>
This equation will look like:<br><br>
y = (0.04576465 * TV + 0.18853002 * Radio - 0.00103749 * Newspaper) + 2.93888937<br>

In [26]:
# confideance score
lr2.score(data_x, y)

0.8972106381789522

ols(ordinary least squares)<br>


In [27]:
# statistical model approach
# this will give statical way of doing impect analysis of model 
# statistical analysis to find out which feature to use and which not
import statsmodels.formula.api as smf
lm = smf.ols(formula="Sales ~ TV + Radio", data=df).fit()
lm.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  Sales   R-squared:                       0.897
Model:                            OLS   Adj. R-squared:                  0.896
Method:                 Least Squares   F-statistic:                     859.6
Date:                Tue, 26 Apr 2022   Prob (F-statistic):           4.83e-98
Time:                        10:11:20   Log-Likelihood:                -386.20
No. Observations:                 200   AIC:                             778.4
Df Residuals:                     197   BIC:                             788.3
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      2.9211      0.294      9.919      0.000       2.340       3.502
TV             0.0458      0.001     32.909      0.000       0.043       0.048
Radio          0.1880      0.008     23.382      0.000       0.172       0.204
==============================================================================
Omnibus:                       60.022   Durbin-Watson:                   2.081
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              148.679
Skew:                          -1.323   Prob(JB):                     5.19e-33
Kurtosis:                       6.292   Cond. No.                         425.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [28]:
import statsmodels.formula.api as smf
lm = smf.ols(formula="Sales ~ TV", data=df).fit()
lm.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  Sales   R-squared:                       0.612
Model:                            OLS   Adj. R-squared:                  0.610
Method:                 Least Squares   F-statistic:                     312.1
Date:                Tue, 26 Apr 2022   Prob (F-statistic):           1.47e-42
Time:                        10:11:20   Log-Likelihood:                -519.05
No. Observations:                 200   AIC:                             1042.
Df Residuals:                     198   BIC:                             1049.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      7.0326      0.458     15.360      0.000       6.130       7.935
TV             0.0475      0.003     17.668      0.000       0.042       0.053
==============================================================================
Omnibus:                        0.531   Durbin-Watson:                   1.935
Prob(Omnibus):                  0.767   Jarque-Bera (JB):                0.669
Skew:                          -0.089   Prob(JB):                        0.716
Kurtosis:                       2.779   Cond. No.                         338.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [29]:
a = smf.ols("Sales ~ TV + Radio + Newspaper", data=df).fit()
a.summary()
# check newspaper p value

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  Sales   R-squared:                       0.897
Model:                            OLS   Adj. R-squared:                  0.896
Method:                 Least Squares   F-statistic:                     570.3
Date:                Tue, 26 Apr 2022   Prob (F-statistic):           1.58e-96
Time:                        10:11:20   Log-Likelihood:                -386.18
No. Observations:                 200   AIC:                             780.4
Df Residuals:                     196   BIC:                             793.6
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      2.9389      0.312      9.422      0.000       2.324       3.554
TV             0.0458      0.001     32.809      0.000       0.043       0.049
Radio          0.1885      0.009     21.893      0.000       0.172       0.206
Newspaper     -0.0010      0.006     -0.177      0.860      -0.013       0.011
==============================================================================
Omnibus:                       60.414   Durbin-Watson:                   2.084
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              151.241
Skew:                          -1.327   Prob(JB):                     1.44e-33
Kurtosis:                       6.332   Cond. No.                         454.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [30]:
a.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  Sales   R-squared:                       0.897
Model:                            OLS   Adj. R-squared:                  0.896
Method:                 Least Squares   F-statistic:                     570.3
Date:                Tue, 26 Apr 2022   Prob (F-statistic):           1.58e-96
Time:                        10:11:20   Log-Likelihood:                -386.18
No. Observations:                 200   AIC:                             780.4
Df Residuals:                     196   BIC:                             793.6
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      2.9389      0.312      9.422      0.000       2.324       3.554
TV             0.0458      0.001     32.809      0.000       0.043       0.049
Radio          0.1885      0.009     21.893      0.000       0.172       0.206
Newspaper     -0.0010      0.006     -0.177      0.860      -0.013       0.011
==============================================================================
Omnibus:                       60.414   Durbin-Watson:                   2.084
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              151.241
Skew:                          -1.327   Prob(JB):                     1.44e-33
Kurtosis:                       6.332   Cond. No.                         454.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [31]:
import pandas as pd
import numpy as np
from pandas_profiling import ProfileReport
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import Ridge, Lasso, RidgeCV, LassoCV, ElasticNet, ElasticNetCV, LinearRegression
from sklearn.model_selection import train_test_split
import statsmodels.api as sm
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [32]:
df = pd.read_csv("Admission_Prediction.csv")
df.head()

Serial No.  GRE Score  TOEFL Score  University Rating  SOP  LOR  CGPA  \
0           1      337.0        118.0                4.0  4.5  4.5  9.65   
1           2      324.0        107.0                4.0  4.0  4.5  8.87   
2           3        NaN        104.0                3.0  3.0  3.5  8.00   
3           4      322.0        110.0                3.0  3.5  2.5  8.67   
4           5      314.0        103.0                2.0  2.0  3.0  8.21   

   Research  Chance of Admit  
0         1             0.92  
1         1             0.76  
2         1             0.72  
3         1             0.80  
4         0             0.65

In [33]:
pf = ProfileReport(df)
pf.to_widgets()

Summarize dataset:   0%|          | 0/22 [00:00<?, ?it/s]

Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render widgets:   0%|          | 0/1 [00:00<?, ?it/s]

In [34]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 500 entries, 0 to 499
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Serial No.         500 non-null    int64  
 1   GRE Score          485 non-null    float64
 2   TOEFL Score        490 non-null    float64
 3   University Rating  485 non-null    float64
 4   SOP                500 non-null    float64
 5   LOR                500 non-null    float64
 6   CGPA               500 non-null    float64
 7   Research           500 non-null    int64  
 8   Chance of Admit    500 non-null    float64
dtypes: float64(7), int64(2)
memory usage: 35.3 KB


In [35]:
df["GRE Score"].fillna(df["GRE Score"].mean(), inplace=True)

In [36]:
df["TOEFL Score"].fillna(df["TOEFL Score"].mean(), inplace=True)

In [37]:
df["University Rating"].fillna(df["University Rating"].mean(), inplace=True)

In [38]:
df.isnull().sum().sum()

0

In [39]:
df.head()

Serial No.   GRE Score  TOEFL Score  University Rating  SOP  LOR  CGPA  \
0           1  337.000000        118.0                4.0  4.5  4.5  9.65   
1           2  324.000000        107.0                4.0  4.0  4.5  8.87   
2           3  316.558763        104.0                3.0  3.0  3.5  8.00   
3           4  322.000000        110.0                3.0  3.5  2.5  8.67   
4           5  314.000000        103.0                2.0  2.0  3.0  8.21   

   Research  Chance of Admit  
0         1             0.92  
1         1             0.76  
2         1             0.72  
3         1             0.80  
4         0             0.65

In [40]:
df.drop(columns=["Serial No."], inplace=True)

In [41]:
df

GRE Score  TOEFL Score  University Rating  SOP  LOR  CGPA  Research  \
0    337.000000        118.0                4.0  4.5  4.5  9.65         1   
1    324.000000        107.0                4.0  4.0  4.5  8.87         1   
2    316.558763        104.0                3.0  3.0  3.5  8.00         1   
3    322.000000        110.0                3.0  3.5  2.5  8.67         1   
4    314.000000        103.0                2.0  2.0  3.0  8.21         0   
..          ...          ...                ...  ...  ...   ...       ...   
495  332.000000        108.0                5.0  4.5  4.0  9.02         1   
496  337.000000        117.0                5.0  5.0  5.0  9.87         1   
497  330.000000        120.0                5.0  4.5  5.0  9.56         1   
498  312.000000        103.0                4.0  4.0  5.0  8.43         0   
499  327.000000        113.0                4.0  4.5  4.5  9.04         0   

     Chance of Admit  
0               0.92  
1               0.76  
2               0.72  
3               0.80  
4               0.65  
..               ...  
495             0.87  
496             0.96  
497             0.93  
498             0.73  
499             0.84  

[500 rows x 8 columns]

In [42]:
y = df[["Chance of Admit"]]
x = df.drop(columns=["Chance of Admit"])

In [43]:
scaler = StandardScaler()

In [44]:
arr = scaler.fit_transform(x)

In [45]:
arr

array([[ 1.84274116e+00,  1.78854223e+00,  7.78905651e-01, ...,
         1.09894429e+00,  1.77680627e+00,  8.86405260e-01],
       [ 6.70814288e-01, -3.10581135e-02,  7.78905651e-01, ...,
         1.09894429e+00,  4.85859428e-01,  8.86405260e-01],
       [ 5.12433309e-15, -5.27312752e-01, -1.07876604e-01, ...,
         1.73062093e-02, -9.54042814e-01,  8.86405260e-01],
       ...,
       [ 1.21170361e+00,  2.11937866e+00,  1.66568791e+00, ...,
         1.63976333e+00,  1.62785086e+00,  8.86405260e-01],
       [-4.10964364e-01, -6.92730965e-01,  7.78905651e-01, ...,
         1.63976333e+00, -2.42366993e-01, -1.12815215e+00],
       [ 9.41258951e-01,  9.61451165e-01,  7.78905651e-01, ...,
         1.09894429e+00,  7.67219636e-01, -1.12815215e+00]])

In [46]:
df1 = pd.DataFrame(arr)
df1

0         1         2         3         4         5         6
0    1.842741e+00  1.788542  0.778906  1.137360  1.098944  1.776806  0.886405
1    6.708143e-01 -0.031058  0.778906  0.632315  1.098944  0.485859  0.886405
2    5.124333e-15 -0.527313 -0.107877 -0.377773  0.017306 -0.954043  0.886405
3    4.905178e-01  0.465197 -0.107877  0.127271 -1.064332  0.154847  0.886405
4   -2.306679e-01 -0.692731 -0.994659 -1.387862 -0.523513 -0.606480 -1.128152
..            ...       ...       ...       ...       ...       ...       ...
495  1.392000e+00  0.134360  1.665688  1.137360  0.558125  0.734118  0.886405
496  1.842741e+00  1.623124  1.665688  1.642404  1.639763  2.140919  0.886405
497  1.211704e+00  2.119379  1.665688  1.137360  1.639763  1.627851  0.886405
498 -4.109644e-01 -0.692731  0.778906  0.632315  1.639763 -0.242367 -1.128152
499  9.412590e-01  0.961451  0.778906  1.137360  1.098944  0.767220 -1.128152

[500 rows x 7 columns]

In [47]:
pf1 = df1.profile_report()
pf1.to_widgets()

Summarize dataset:   0%|          | 0/20 [00:00<?, ?it/s]

Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render widgets:   0%|          | 0/1 [00:00<?, ?it/s]

In [48]:
df1.describe()

0             1             2             3             4  \
count  5.000000e+02  5.000000e+02  5.000000e+02  5.000000e+02  5.000000e+02   
mean   4.350520e-15  9.419132e-16  5.608847e-16  2.926548e-16 -1.332268e-17   
std    1.001002e+00  1.001002e+00  1.001002e+00  1.001002e+00  1.001002e+00   
min   -2.394225e+00 -2.512331e+00 -1.881441e+00 -2.397950e+00 -2.686789e+00   
25%   -6.814090e-01 -6.927310e-01 -9.946589e-01 -8.828175e-01 -5.235128e-01   
50%    5.124333e-15 -3.105811e-02 -1.078766e-01  1.272712e-01  1.730621e-02   
75%    6.708143e-01  7.960330e-01  7.789057e-01  6.323155e-01  5.581253e-01   
max    2.113186e+00  2.119379e+00  1.665688e+00  1.642404e+00  1.639763e+00   

                  5             6  
count  5.000000e+02  5.000000e+02  
mean   3.091971e-15 -2.202682e-16  
std    1.001002e+00  1.001002e+00  
min   -2.940115e+00 -1.128152e+00  
25%   -7.430227e-01 -1.128152e+00  
50%   -2.720919e-02  8.864053e-01  
75%    7.672196e-01  8.864053e-01  
max    2.223672e+00  8.864053e-01

In [1]:
from statsmodels.stats.outliers_influence import variance_inflation_factor

In [50]:
arr.shape[1]

7

In [51]:
[variance_inflation_factor(arr, i) for i in range(arr.shape[1])]

[4.153267572225826,
 3.792866110594648,
 2.5087682422787627,
 2.7757495092534965,
 2.0373076624897517,
 4.651669561154733,
 1.4593106786827286]

In [52]:
vif_df = pd.DataFrame()
vif_df["vif"] = [variance_inflation_factor(arr, i) for i in range(arr.shape[1])]

In [53]:
vif_df

vif
0  4.153268
1  3.792866
2  2.508768
3  2.775750
4  2.037308
5  4.651670
6  1.459311

In [54]:
x_train, x_test, y_train, y_test = train_test_split(arr, y, test_size=0.25, random_state=345)

In [55]:
x_train

array([[-4.10964364e-01, -6.92730965e-01,  7.78905651e-01, ...,
         1.63976333e+00, -2.42366993e-01, -1.12815215e+00],
       [-1.67303946e+00, -1.02356739e+00,  7.78905651e-01, ...,
         1.09894429e+00, -1.46711143e+00,  8.86405260e-01],
       [-4.10964364e-01,  1.34360100e-01, -1.07876604e-01, ...,
        -5.23512832e-01, -7.68609886e-02, -1.12815215e+00],
       ...,
       [ 4.90517846e-01,  1.29228759e+00,  1.66568791e+00, ...,
         1.09894429e+00,  1.29683885e+00,  8.86405260e-01],
       [-1.67303946e+00, -1.68524024e+00,  3.93810431e-16, ...,
        -5.23512832e-01, -2.26154025e+00, -1.12815215e+00],
       [ 1.75259294e+00,  1.95396044e+00,  1.66568791e+00, ...,
         1.73062093e-02,  2.02506527e+00,  8.86405260e-01]])

In [56]:
lr = LinearRegression()
lr.fit(x_train, y_train)

LinearRegression()

In [57]:
lr.predict([[337.000000,118.0,4.0,4.5,4.5,9.65,1]])

array([[9.86354874]])

In [58]:
df

GRE Score  TOEFL Score  University Rating  SOP  LOR  CGPA  Research  \
0    337.000000        118.0                4.0  4.5  4.5  9.65         1   
1    324.000000        107.0                4.0  4.0  4.5  8.87         1   
2    316.558763        104.0                3.0  3.0  3.5  8.00         1   
3    322.000000        110.0                3.0  3.5  2.5  8.67         1   
4    314.000000        103.0                2.0  2.0  3.0  8.21         0   
..          ...          ...                ...  ...  ...   ...       ...   
495  332.000000        108.0                5.0  4.5  4.0  9.02         1   
496  337.000000        117.0                5.0  5.0  5.0  9.87         1   
497  330.000000        120.0                5.0  4.5  5.0  9.56         1   
498  312.000000        103.0                4.0  4.0  5.0  8.43         0   
499  327.000000        113.0                4.0  4.5  4.5  9.04         0   

     Chance of Admit  
0               0.92  
1               0.76  
2               0.72  
3               0.80  
4               0.65  
..               ...  
495             0.87  
496             0.96  
497             0.93  
498             0.73  
499             0.84  

[500 rows x 8 columns]

In [59]:
test1 = scaler.transform([[337.000000,118.0,4.0,4.5,4.5,9.65,1]])
test1

/home/calkikhunt/.local/lib/python3.8/site-packages/sklearn/base.py:450: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names
  warnings.warn(


array([[1.84274116, 1.78854223, 0.77890565, 1.13735981, 1.09894429,
        1.77680627, 0.88640526]])

In [60]:
lr.predict(test1)

array([[0.95377928]])

In [61]:
lr.score(x_test, y_test)

0.7476466381844957

In [62]:
def adj_r2(x, y):
    r2 = lr.score(x, y)
    n = x.shape[0]
    p = x.shape[1]
    adjusted_r2 = 1-(1-r2)*(n-1)/(n-p-1)
    return adjusted_r2

In [63]:
adj_r2(x_test, y_test)

0.7325485738023716

In [64]:
lr.coef_

array([[0.02006269, 0.01295679, 0.00978097, 0.00451013, 0.01626859,
        0.07320679, 0.01332327]])

In [65]:
lr.intercept_

array([0.72112471])

In [66]:
# lasso regularization
lassocv = LassoCV(cv=10, max_iter=200000, normalize=True)
lassocv.fit(x_train, y_train)

/home/calkikhunt/.local/lib/python3.8/site-packages/sklearn/linear_model/_coordinate_descent.py:1571: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


LassoCV(cv=10, max_iter=200000, normalize=True)

In [69]:
lassocv.alpha_

6.608444189659907e-06

In [70]:
lasso = Lasso(alpha=lassocv.alpha_)
lasso.fit(x_train, y_train)

Lasso(alpha=6.608444189659907e-06)

In [71]:
lasso.score(x_test, y_test)

0.7476607382238118

In [76]:
# ridge regularization
ridgecv = RidgeCV(alphas=np.random.uniform(0,10,50), cv=5, normalize=True)
ridgecv.fit(x_train, y_train)

RidgeCV(alphas=array([7.61647282, 0.588858  , 8.65394222, 2.72432137, 0.20126482,
       7.0545374 , 0.68363427, 7.67303592, 5.7530052 , 8.49702974,
       4.75568241, 7.48115761, 5.05684881, 6.19774382, 9.53913185,
       7.70163113, 9.24313104, 1.81706178, 6.7056828 , 4.47250879,
       6.49634693, 4.8529727 , 6.23734129, 3.56500019, 5.01936134,
       8.10131525, 5.74713971, 9.85108724, 2.20891029, 1.33250446,
       7.81653578, 2.24266722, 9.94598104, 9.06277933, 2.11767754,
       9.19926663, 5.7623225 , 7.89337294, 6.02341989, 8.61283574,
       9.49644874, 7.67614676, 2.99456412, 8.18256438, 3.91518139,
       3.25021491, 0.20257201, 1.4732118 , 4.76127283, 8.61174862]),
        cv=5, normalize=True)

In [77]:
ridgecv.alpha_

0.20126482395621315

In [78]:
ridge = Ridge(alpha=ridgecv.alpha_)
ridge.fit(x_train, y_train)

Ridge(alpha=0.20126482395621315)

In [79]:
ridge.score(x_test, y_test)

0.7476519296682583

In [84]:
# Elastic net regularization
elastic_cv = ElasticNetCV(cv=5)
elastic_cv.fit(x_train, y_train)

/home/calkikhunt/.local/lib/python3.8/site-packages/sklearn/linear_model/_coordinate_descent.py:1571: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


ElasticNetCV(cv=5)

In [85]:
elastic_cv.alpha_

0.0002564997621834137

In [86]:
elastic_cv.l1_ratio_

0.5

In [87]:
elastic = ElasticNet(alpha=elastic_cv.alpha_, l1_ratio=elastic_cv.l1_ratio)
elastic.fit(x_train, y_train)

ElasticNet(alpha=0.0002564997621834137)

In [88]:
elastic.score(x_test, y_test)

0.7479204103015531

In [89]:
from sklearn.inspection import permutation_importance

permutation_importance()

TypeError: permutation_importance() missing 3 required positional arguments: 'estimator', 'X', and 'y'